python/note_input_widget/widget.py

```diff
@@ -46,14 +49,18 @@ class NoteInputWidget(QtGui.QWidget):
     data_updated = QtCore.Signal()
     close_clicked = QtCore.Signal()
 
+    # Emitted when a new note or reply will be created. The creation request id
+    # is sent and will match the id sent with the entity_created signal.
+    # Todo: This has functional overlap with the pre_submit_callback...resolve that...
+    new_entity_requested = QtCore.Signal(str, int)
```

Comment:
@gomezll  I added these extra signals and functionality before I realized there is already a mechanism that supplies similar functionality - the `pre_submit_callback`. Imo the existing callback is a bit odd since it passes the `NoteInputWidget` object as its parameter which feels a bit arbitrary, and some work would be needed to use that one since the pending request UID would need to be exposed through a function and that also seems a bit random.

I took the approach of adding new signals specific for the exact event type that we're trying to track (shotgun note creations and external user creations (in activity_stream.py), which seems nice and explicit but also makes things a bit busier.

Interested to see what you think.